pushy/src/main/java/com/turo/pushy/apns/proxy/HttpProxyHandlerFactory.java

```diff
@@ -41,6 +47,53 @@
     private final String username;
     private final String password;
 
+    /**
+     * Checks the system default proxies to determine if an {@link HttpProxyHandlerFactory}
+     * is required for the given host or not, returning an instance or null for each case.
+     *
+     * @param apnsHost the APNs host to check e.g. "api.push.apple.com".
+     * @return an {@link HttpProxyHandlerFactory} if one is required, or <code>null</code> if not.
+     * @throws URISyntaxException if <code>apnsHost</code> is malformed in some way.
+     *
+     * @see #fromSystemProxies()
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#Proxies">The Java documentation on proxies.</a>
+     * @since 0.14
+     */
+    public static HttpProxyHandlerFactory fromSystemProxies(final String apnsHost)
+            throws URISyntaxException {
+
+        URI hostUri = new URI("https", apnsHost, null, null);
+        SocketAddress proxyAddress = ProxyLocator.getProxyAddressForUri(hostUri, Proxy.Type.HTTP);
+
+        if (proxyAddress != null) {
+            return new HttpProxyHandlerFactory(proxyAddress);
+
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Checks the system default proxies to determine if an {@link HttpProxyHandlerFactory}
+     * is required for the production APNs host or not, returning an instance or null for each case.
+     *
+     * @return an {@link HttpProxyHandlerFactory} if one is required, or <code>null</code> if not.
+     *
+     * @see #fromSystemProxies(String)
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#Proxies">The Java documentation on proxies.</a>
+     * @since 0.14
+     */
+    public static HttpProxyHandlerFactory fromSystemProxies() {
```

Comment:
I definitely appreciate the thought here, but we've been fairly deliberate in not providing any "choose a server by default" mechanisms elsewhere in Pushy. For consistency, I think it might be best to drop this method.

As an alternative, it might be helpful to provide a `@see` tag on `fromSystemProperties(String)` that points to the pre-defined server constants in `ApnsClientBuilder`.

Reply:
I'll drop it; I'm using the other one myself and I definitely agree with explicit being better than implicit.